docs/table-migration.md

```diff
@@ -0,0 +1,89 @@
+ - (the "License"); you may not use this file except in compliance with
+ - the License.  You may obtain a copy of the License at
+ -
+ -   http://www.apache.org/licenses/LICENSE-2.0
+ -
+ - Unless required by applicable law or agreed to in writing, software
+ - distributed under the License is distributed on an "AS IS" BASIS,
+ - WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ - See the License for the specific language governing permissions and
+ - limitations under the License.
+ -->
+# Table Migration
+Apache Iceberg supports converting existing tables in other formats to Iceberg tables. This section introduces the general concept of table migration, its approaches, and existing implementations in Iceberg.
+
+## Migration Approaches
+There are two main approaches to perform table migration: CTAS (Create Table As Select) and in-place migration.
+
+### Create-Table-As-Select Migration
+CTAS migration involves creating a new Iceberg table and copying data from the existing table to the new one. This method is preferred when you want to completely cut ties with your old table, ensuring the new table is independent and fully managed by Iceberg.
+However, CTAS migration may require more time to complete and might not be suitable for production use cases where downtime is not acceptable.
+![CTAS Migration](../../../img/iceberg-CTAS.png)
+### In-Place Migration
+In-place migration retains the existing data files but adds Iceberg metadata on top of them. This approach is faster and does not require copying data, making it more suitable for production use cases.
+![In-Place Migration](../../../img/iceberg-In-place.png)
+## In-Place Migration Actions
+Apache Iceberg primarily supports the in-place migration approach, which includes three important actions:
+
+1. Snapshot Table
+2. Migrate Table
+3. Add Files
+
+### Snapshot Table
+The Snapshot Table action creates a new iceberg table with the same schema and partitioning as the source table, leaving the source table unchanged during and after the action.
+
+**Step 1:** Create a new Iceberg table with the same metadata (schema, partition spec, etc.) as the source table
+![Snapshot Table Step 1](../../../img/iceberg-snapshotaction-step1.png)
+
+**Step 2:** Commit all data files across all partitions to the new Iceberg table. The source table remains unchanged.
+![Snapshot Table Step 2](../../../img/iceberg-snapshotaction-step2.png)
+### Migrate Table
+The Migrate Table action also creates a new Iceberg table with the same schema and partitioning as the source table. However, during the action execution, it locks and drops the source table from the catalog.
+Consequently, Migrate Table requires all readers and writers working on the source table to be stopped before the action is performed.
+
+**Step 1:** Stop all readers and writers interacting with the source table
+![Migrate Table Step 1](../../../img/iceberg-migrateaction-step1.png)
```

Comment:
These graphs are supposed to be located in https://github.com/apache/iceberg-docs/tree/main/iceberg-theme/static/img. Not sure if I need to open a separate PR in `iceberg-doc` to add these graphs.

But before that, I would like to put a preview of the page with pictures here

[table_migration_preview.pdf](https://github.com/apache/iceberg/files/11083574/table_migration_preview.pdf)

Reply:
Yes you can open a separated PR for that in iceberg-doc

Reply:
Opened a PR here: https://github.com/apache/iceberg-docs/pull/220